src/loghandler.cpp

```diff
@@ -233,26 +222,49 @@ void LogHandler::writeLogs(QTextStream& out) {
 void LogHandler::cleanupLogs() {
   QMutexLocker lock(&s_mutex);
 
-  if (!s_instance || s_instance->m_logFileName.isEmpty()) {
+  if (!s_instance || !s_instance->m_logFile) {
     return;
   }
 
+  QString logFileName = s_instance->m_logFile->fileName();
   s_instance->closeLogFile(lock);
 
   {
-    QFile file(s_instance->m_logFileName);
+    QFile file(logFileName);
     file.remove();
   }
 
   s_instance->openLogFile(lock);
 }
 
+// static
+void LogHandler::setLocation(const QString& path) {
+  s_location = path;
+
+  if (s_instance && s_instance->m_logFile) {
+    cleanupLogs();
```

Comment:
Maybe pass `lock` here and have 2 `cleanupLogs()` methods, 1 with the proofOfLock and 1 without it.